### ISSUE TITLE
Flycheck ignores eslintrc rules

### ISSUE THREAD

**Issue by shubham-cpp** [flycheck/flycheck#1941]
Thank you for taking the time to report an issue and improve Flycheck.  This template is for **actual bugs you observed**.  If you have trouble setting up Flycheck, or if you have a question, please use the relevant issue template instead.

## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description
Flycheck is ignoring the .eslintrc rules and provides false errors

## Steps to reproduce
1. Open file jsx/tsx file with .eslintrc in root folder
2. Call global-flycheck-mode
3. See error

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior
Flycheck should respect .eslintrc rules and not provide false errors

## Screenshots
![image](https://user-images.githubusercontent.com/31388020/167253304-5bf7fc2c-3885-42b6-b0bf-1355d8de5eb4.png)

## System configuration
<details>

<summay>M-x flycheck-verify-setup</summay>

```
Syntax checkers for buffer App.tsx in rjsx-mode:

First checker to run:

  javascript-eslint
    - may enable:  yes
    - executable:  Found at /home/shubham/.local/share/nvm/versions/node/v18.0.0/bin/eslint
    - config file: found

Checkers that are compatible with this mode, but will not run until properly configured:

  javascript-jshint (manually disabled) enable
    - may enable:         no
    - executable:         Not found
    - configuration file: Not found

  javascript-standard (manually disabled) enable
    - may enable: no
    - executable: Not found

Flycheck Mode is enabled.  Use M-x universal-argument C-c ! x to
enable disabled checkers.

--------------------

Flycheck version: 33snapshot (package: 20220504.830)
Emacs version:    29.0.50
System:           x86_64-pc-linux-gnu
Window system:    pgtk

```

</details>

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes
My minimal emacs config to reproduce this issue

<details>

```elisp
(require 'package)
(add-to-list 'package-archives '("gnu"   . "https://elpa.gnu.org/packages/"))
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/"))
(package-initialize)

(unless (package-installed-p 'use-package)
  (package-refresh-contents)
  (package-install 'use-package))

(eval-and-compile
  (setq use-package-always-ensure t
        use-package-expand-minimally t))

(load-theme 'leuven-dark)

(use-package rjsx-mode
  :mode ("\\.tsx\\'" "\\.jsx\\'"))

(use-package flycheck
  :config
  (with-eval-after-load 'flycheck
    (setq-default flycheck-disabled-checkers
		  (append flycheck-disabled-checkers
			  '(javascript-jshint json-jsonlist)))
    (push 'web-mode (flycheck-checker-get 'javascript-eslint 'modes)))
	(global-flycheck-mode))


```

</details>

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by shubham-cpp** [comment]
Sorry this caused by js2-mode. This resolved it
```elisp
(setq js2-strict-missing-semi-warning nil)
```
